src/script/descriptor.cpp

```diff
@@ -275,29 +288,37 @@ class BIP32PubkeyProvider final : public PubkeyProvider
     BIP32PubkeyProvider(const CExtPubKey& extkey, KeyPath path, DeriveType derive) : m_extkey(extkey), m_path(std::move(path)), m_derive(derive) {}
     bool IsRange() const override { return m_derive != DeriveType::NO; }
     size_t GetSize() const override { return 33; }
-    bool GetPubKey(int pos, const SigningProvider& arg, CPubKey* key, KeyOriginInfo& info) const override
+    bool GetPubKey(int pos, const SigningProvider& arg, CPubKey* key, KeyOriginInfo& info, const DescriptorCache* read_cache = nullptr, DescriptorCache* write_cache = nullptr) const override
     {
+        KeyOriginInfo info_out;
```

Comment:
We want to avoid modifying `info` in case later steps fail.

Reply:
might be worth a comment(took me a bit to convince myself what it was doing)

Reply:
Then we need another test, because everything passed when I changed that.

Reply:
I added a comment. I don't think a test is useful. Everything still passes because `ExpandHelper` does the same thing. Just keeping it for belt-and-suspenders.